UI/src/view/linear_asset/laneModellingStyle.js

```diff
@@ -32,18 +32,33 @@
       }
 
       var linearAssetsWithType = _.flatten(_.map(relevantLinears, function(linearAsset) {
-        var hasAsset = me.hasValue(linearAsset);
+        var hasAsset = me.hasValue(linearAsset) || !_.isUndefined(linearAsset.isViewOnly);
         var type = {type: 'line'};
 
         if (isRoadlink || (!_.isUndefined(linearAsset.points) && !isRoadlink && linearAsset.selectedLinks.length == 1)) {
-          return _.merge({}, linearAsset, {hasAsset: hasAsset}, type);
+         if (!_.isEmpty(linearAsset.selectedLinks)) {
+           var link = _.find(linearAsset.selectedLinks, {'linkId': linearAsset.linkId});
+           var options = {
+           hasAsset: hasAsset,
+           notSelectable: !hasAsset,
+           roadPartNumber: link.roadPartNumber,
+           startAddrMValue: link.startAddrMValue,
+           endAddrMValue: link.endAddrMValue,
+           isSelected: true
+           };
+           return _.merge({}, linearAsset, type, options);
+         } else
+           return _.merge({}, linearAsset, {hasAsset: hasAsset, notSelectable: !hasAsset}, type);
```

Comment:
Suggestion:

```js

if (isRoadlink || (!_.isUndefined(linearAsset.points) && !isRoadlink && linearAsset.selectedLinks.length == 1)) {

var hasAssetAndSelectable = { hasAsset: hasAsset, notSelectable: !hasAsset };

    if (!_.isEmpty(linearAsset.selectedLinks)) {
           var link = _.find(linearAsset.selectedLinks, {'linkId': linearAsset.linkId});
           var options = {
           roadPartNumber: link.roadPartNumber,
           startAddrMValue: link.startAddrMValue,
           endAddrMValue: link.endAddrMValue,
           isSelected: true
           };
           return _.merge({}, linearAsset, hasAssetAndSelectable , type, options);
         } else
           return _.merge({}, linearAsset, hasAssetAndSelectable , type);

```

Reply:
Done, merged three suggestions of this nature